kuksa-sdk/src/main/kotlin/org/eclipse/kuksa/extension/VssSpecificationCopyExtension.kt

```diff
@@ -137,17 +148,38 @@ fun <T : VssSpecification> T.copy(
         .find { it.vssPath == vssPath } ?: return this
 
     val updatedVssProperty = vssProperty.copy(updatedValue)
-    val relevantChildren = findHeritageLine(vssProperty).toMutableList()
 
-    // Replace the last specification (Property) with the changed one
-    val updatedVssSpecification: T = if (relevantChildren.isNotEmpty()) {
-        relevantChildren.removeLast()
-        relevantChildren.add(updatedVssProperty)
+    // Same property with no heirs, no deep copy is needed
+    if (this.vssPath == updatedVssProperty.vssPath) return updatedVssProperty as T
 
-        this.deepCopy(relevantChildren)
-    } else {
-        updatedVssProperty as T // The property must be T since no children are available
-    }
+    return deepCopy(0, updatedVssProperty)
+}
 
-    return updatedVssSpecification
+/**
+ * Convenience operator for [copy] with a value [T].
+ */
+operator fun <T : Any> VssProperty<T>.plusAssign(value: T) {
+    copy(value)
+}
+
+/**
+ * Convenience operator for [copy] with a value [T].
+ */
+operator fun <T : Any> VssProperty<T>.plus(value: T): VssProperty<T> {
+    return copy(value)
+}
```

Comment:
Completely agree. The plus operator is not a good choice here.

Reply:
Converted the operator to "invoke". My train off thought is here: It looks similar to calling the data class constructor with all the children so it may at least be a little convenient and intuitiv. :)
- VssVehicle(body: VssBody) // new construct
- vssVehicle(vssAirbag) // copy

If you are not seeing this then we can remove it again. Maybe I am already too used to it now.^^